algorithms/2.TCGA-MLexample-LungCancer.py

```diff
@@ -0,0 +1,166 @@
+
+# coding: utf-8
+
+# # Create a logistic regression model to predict gene mutation from gene expression data in TCGA
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[3]:
+
+# Parameter Sweep for Hyperparameters
+n_feature_kept = 500
+param_fixed = {
+    'loss': 'log',
+    'penalty': 'elasticnet',
+}
+param_grid = {
+    'alpha': [10 ** x for x in range(-6, 1)],
+    'l1_ratio': [0, 0.05, 0.1, 0.2, 0.5, 0.8, 0.9, 0.95, 1],
+}
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+# 
+# *Here is some [information](https://ghr.nlm.nih.gov/gene/TP53) about TP53*
+
+# ## Load Data
+
+# In[4]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('download', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[5]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('download', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[6]:
+
+# GENES: AKT1, ALK, BRAF, DDR2, EGFR, ERBB2, KRAS, MAP2K1, NRAS, PIK3CA, PTEN, RET, RIT1, ROS1
+
+list_of_tsgenes = [207,238,673,4921,1956,2064,3845,5604,4893,5290,5728,5979,6016,6098]

```

Comment:
> Thank you for the feedback! I used ids listed on the NCBI website. I made sure that the numbers correspond to the Homo sapiens column

That's the correct way to do it. See [this example](https://github.com/cognoma/cancer-data/blob/b054721bbde1ed0abc19f395eb7629728870cb7c/scripts/3.explore-mutations.py#L100-L109) for what I mean by the dictionary. The goal is to make your plot labels show gene symbols rather than entrez GeneIDs.